#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_backend_crash_without_tuner.dpatch by Kees Cook <kees@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Closes LP: #93250.
## DP: http://www.archivesat.com/MythTV_development/thread2049262.htm

@DPATCH@
diff -urNad mythtv-0.20-svn20070122~/programs/mythbackend/mainserver.cpp mythtv-0.20-svn20070122/programs/mythbackend/mainserver.cpp
--- mythtv-0.20-svn20070122~/programs/mythbackend/mainserver.cpp	2007-01-22 01:08:40.000000000 -0800
+++ mythtv-0.20-svn20070122/programs/mythbackend/mainserver.cpp	2007-03-23 22:47:40.616622082 -0700
@@ -2951,10 +2951,22 @@
     QMap<int, EncoderLink *>::Iterator iter = encoderList->find(recnum);
     if (iter == encoderList->end())
     {
-        VERBOSE(VB_IMPORTANT, "MainServer: " +
-                QString("HandleRemoteEncoder(cmd %1) ").arg(slist[1]) +
-                QString("Unknown encoder: %1, exiting").arg(recnum));
-        exit(BACKEND_BUGGY_EXIT_UNKNOWN_ENC);
+        // Encoder not found
+        QString command = slist[1];
+        if (command == "GET_STATE")
+        {
+           QStringList retlist;
+           retlist << QString::number((int)kState_Error);
+           SendResponse(pbssock, retlist);
+           return;
+        }
+        else
+        {
+           VERBOSE(VB_IMPORTANT, "MainServer: " +
+                   QString("HandleRemoteEncoder(cmd %1) ").arg(slist[1]) +
+                   QString("Unknown encoder: %1, exiting").arg(recnum));
+           exit(BACKEND_BUGGY_EXIT_UNKNOWN_ENC);
+        }
     }
 
     EncoderLink *enc = iter.data();
